NuRadioReco/modules/RNO_G/triggerBoardResponse.py

```diff
@@ -126,58 +99,71 @@ def apply_adc_gain(self, station, det, trigger_channels, avg_vrms=None):
         avg_rms : float (default: None)
             The Vrms of the trigger channels including the trigger board filters
             If set to `None`, this will be estimated using the waveforms
+        gain_values : list (default: None)
+            If set these will be applied to the channel. Otherwise gains are recalculated.
 
         Returns
         -------
-        vrms_after_gain : float
+        vrms_after_gain : list
             the RMS voltage of the waveforms after the gain has been applied
-
         ideal_vrms: float
             the ideal vrms, as measured on the ADC capacitors
-
+        ret_gain_values: list
+            gain values applied to each channel
         """
 
         if avg_vrms is None:
-            avg_vrms = self.get_avg_vrms(station, trigger_channels)
+            avg_vrms = self.get_vrms(station, trigger_channels)
 
         self.logger.debug("Applying gain at ADC level")
 
         if not hasattr(avg_vrms, "__len__"):
             avg_vrms = np.full_like(trigger_channels, avg_vrms, dtype=float)
 
         vrms_after_gain = []
+        ret_gain_values = []
+
         for channel_id, vrms in zip(trigger_channels, avg_vrms):
             det_channel = det.get_channel(station.get_id(), channel_id)
 
             noise_bits = det_channel["trigger_adc_noise_nbits"]
             total_bits = det_channel["trigger_adc_nbits"]
-            volts_per_adc = self._adc_input_range / (2 ** total_bits - 1)
-            ideal_vrms = volts_per_adc * (2 ** (noise_bits - 1))
-
-            msg = f"\t Ch: {channel_id}\t Target Vrms: {ideal_vrms / units.mV:0.3f} mV"
-            msg += f"\t V/ADC: {volts_per_adc / units.mV:0.3f} mV"
-            self.logger.debug(msg)
-
-            # find the ADC gain from the possible values that makes the realized
-            # vrms closest-to-but-greater-than the ideal value
-            amplified_vrms_values = vrms * self._triggerBoardAmplifications
-            mask = amplified_vrms_values > ideal_vrms
-            if np.any(mask):
-                gain_to_use = self._triggerBoardAmplifications[mask][0]
-                vrms_after_gain.append(amplified_vrms_values[mask][0])
+            volts_per_adc = self._adc_input_range / (2 ** total_bits)
```

Comment:
there is a `-1` missing (not in the exponent). If you have a 8bit system you want to divide by 2 ** 8 -1 = 255